### PR TITLE
feat(user-profile): Adding user device tokens admin and lib client 

### PR DESCRIPTION
**Pull request by rafnarnason**
# ...

Attach a link to issue if relevant

## What

make devicetokens accessible internally via admin scoped endpoint and lib client

## Why

so the notifications system can read devicetokens for push notifications

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/clients/user-profile/src/lib/userProfileClient.module.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { Module } from '@nestjs/common'
+import { UserProfileApi } from '../../gen/fetch'
+
+import { ApiConfiguration } from './apiConfiguration'
+// import { exportedApis } from './apis'
```
Comment (eirikurn) [inline libs/clients/user-profile/src/lib/userProfileClient.module.ts:5]: Remove?

File: libs/clients/user-profile/src/lib/apiConfiguration.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { createEnhancedFetch } from '@island.is/clients/middlewares'
+import { ConfigType } from '@island.is/nest/config'
+import { UserProfileApi, Configuration } from '../../gen/fetch'
+
+import { UserProfileClientConfig } from './userProfileClient.config'
+
+export const ApiConfiguration = {
```
Comment (eirikurn) [inline libs/clients/user-profile/src/lib/apiConfiguration.ts:7]: Maybe `UserProfileApiProvider`?

File: libs/clients/user-profile/README.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+@Injectable()
+export class SomeService {
+  constructor(
+    @Inject(UserProfileApi)
+    private readonly userProfileApi: UserProfileApi,
+  ) {}
+
+  async getStuff()
+      const clientAuthToken = await acquireAuthToken()
+      const currentAuth = convertTokenToAuth(clientAuthToken)
```
Comment (eirikurn) [inline libs/clients/user-profile/README.md:45]: Example should either show CurrentUser object or #6057.

File: apps/services/user-profile/src/openApi.ts [hunk 4]
```diff
@@ -15,6 +15,7 @@ export const openApi = new DocumentBuilder()
         scopes: {
           ['openid']: '',
           ['profile']: '',
+          [UserProfileScope.admin]: '',
```
Comment (eirikurn) [inline apps/services/user-profile/src/openApi.ts:18]: Description?

File: apps/services/user-profile/src/app/user-profile/userProfile.controller.ts [hunk 5]
```diff
@@ -397,3 +379,24 @@ export class UserProfileController {
     }
   }
 }
+
+// this controller separate for guards and scopes logic - idsauthguard vs idsuserguard
+@UseGuards(IdsAuthGuard, ScopesGuard)
+@ApiTags('User Profile')
+@Controller()
+export class UserTokenController {
```
Comment (eirikurn) [inline apps/services/user-profile/src/app/user-profile/userProfile.controller.ts:387]: Should move to separate file?

File: apps/services/user-profile/src/app/user-profile/userProfile.controller.ts [hunk 6]
```diff
@@ -397,3 +379,24 @@ export class UserProfileController {
...
+  constructor(private readonly userProfileService: UserProfileService) {}
+  @ApiOperation({
+    summary: 'admin access - returns a list of user device tokens',
+  })
+  @ApiOkResponse({ type: [UserDeviceTokensDto] })
+  @Scopes(UserProfileScope.admin)
+  @ApiSecurity('oauth2', [UserProfileScope.admin])
+  @Get('userProfile/:nationalId/deviceToken')
```
Comment (eirikurn) [inline apps/services/user-profile/src/app/user-profile/userProfile.controller.ts:395]: API design guide: "/device-tokens"?

**Review by eirikurn — CHANGES_REQUESTED** [review]

File: apps/services/user-profile/src/app/user-profile/userToken.controller.ts [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+  Controller,
+  Get,
+
+  Param,
+
+  UseGuards,
+
+} from '@nestjs/common'
+import {
+
```
Comment (eirikurn) [inline apps/services/user-profile/src/app/user-profile/userToken.controller.ts:17]: Strange that format doesn't catch whitespace like this 🤷

**Review by eirikurn — APPROVED** [review]
👍

**Review by disaerna — APPROVED** [review]
Looks good to me 👍

**Pull request merged**